<a href="https://colab.research.google.com/github/BenoitProst/Genealogy_map/blob/master/Table_mortalit%C3%A9_Titanic.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

In [ ]:
chemin_fichier = "/content/drive/MyDrive/Développement compétences/CNAM Master 1 actuariat/ACT102 - Actuariat Vie/titanic_train.csv"

Chargement du fichier Titanic dans un dataframe Pandas

In [ ]:
df = pd.read_csv(chemin_fichier, sep=';')

On remarque que certains ages ne sont pas des entiers sur l'année.
Création d'intervalles d'un an

In [ ]:
bornes_age_par_an = list(range(0, 101))

In [ ]:
df['Age_par_an'] = pd.cut(df['Age'], bins=bornes_age_par_an, right=False)

On compte le nombre de passagers, la population, par age, sexe et classe

In [ ]:
df_exposition = df.groupby(['Age_par_an', 'Sex', 'Pclass']).size().reset_index(name='nombre_passager')

On fait la même approche avec les survivants

In [ ]:
df_survivant = df.groupby(['Age_par_an', 'Sex', 'Pclass'])['Survived'].sum().reset_index()

On fusionne les deux taleaux sur age, sexe et classe

In [ ]:
df_survie = pd.merge(df_exposition, df_survivant, on=['Age_par_an', 'Sex', 'Pclass'], how='inner')

On calcule ainsi le taux de survie $p_{x}$ avec le nombre de survivants sur le nombre de passagers

In [ ]:
df_survie['Px']=df_survie['Survived']/df_survie['nombre_passager']

Regardons la probabilité de survie de Jack, 15ans, voyageant en 3ème classe

In [ ]:
df_survie[df_survie['Age_par_an']== pd.Interval(15, 16, closed='left')]

Age_par_an     Sex  Pclass  nombre_passager  Survived   Px
90   [15, 16)  female       1                1         1  1.0
91   [15, 16)  female       2                0         0  NaN
92   [15, 16)  female       3                3         3  1.0
93   [15, 16)    male       1                0         0  NaN
94   [15, 16)    male       2                0         0  NaN
95   [15, 16)    male       3                1         0  0.0

Sur la ligne 95, on peut voir qu'elle est de 0, qu'importe de quel port il embarque.
Donc si l'Olympic fait naufrage, on peut en déduire que Jack meurt de façon quasi certaine.

Ainsi sur la traversé, la probabilité de décès de Jack est de 10% (= probabilité de naufrage).

Si Jack veut assurer un capital de 100 000€ en cas de décès, la prime pure est **10 000€**.

Regardons maintenant la probabilité de survie de Rose, 17ans, voyageant en 1ère classe

In [ ]:
df_survie[df_survie['Age_par_an']== pd.Interval(17, 18, closed='left')]

Age_par_an     Sex  Pclass  nombre_passager  Survived   Px
102   [17, 18)  female       1                2         2  1.0
103   [17, 18)  female       2                2         2  1.0
104   [17, 18)  female       3                2         1  0.5
105   [17, 18)    male       1                1         1  1.0
106   [17, 18)    male       2                0         0  NaN
107   [17, 18)    male       3                6         0  0.0

Sur la ligne 102, on peut noter que sa probabilité de survie est de 100% en cas de naufrage. Sa prime pure est donc nulle.

Rose a donc **peu d'intérêt** de s'assurer.

Ses résultats sont à relativiser par rapport au faible nombre de personnes dans l'échantillon analysé.